test/uk/gov/hmrc/economiccrimelevyregistration/services/RegistrationValidationServiceSpec.scala

```diff
@@ -375,5 +372,39 @@ class RegistrationValidationServiceSpec extends SpecBase {
           )
         )
     }
+
+    "return errors if the registration for a other entity is invalid" in forAll {
+      (validCharityRegistration: ValidCharityRegistration) =>
+        val invalidRegistration = validCharityRegistration.registration.copy(
+          optOtherEntityJourneyData = None
+        )
+        val result              = service.validateRegistration(invalidRegistration)
+        result.isValid shouldBe false
+        result.leftMap(nec => nec.toList.isEmpty shouldBe false)
+    }
+
+    "return the registration if the registration for a charity is valid" in forAll {
+      (validCharityRegistration: ValidCharityRegistration) =>
+        val result = service.validateRegistration(validCharityRegistration.registration)
+        result shouldBe Valid(Right(validCharityRegistration.registration))
+    }
+
+    "return errors if the registration for a charity is invalid" in forAll {
+      (validCharityRegistration: ValidCharityRegistration) =>
+        val otherEntityJourneyData     = validCharityRegistration.registration.otherEntityJourneyData.copy(
+          charityRegistrationNumber = None
```

Comment:
you can include missing Company registration number in this test as well

Reply:
Ammended test.